### PR TITLE
Add API call to Delete Button

### PR DESCRIPTION
**Pull request by kappanjoe**
## What
- `Delete` button now calls API to delete selected drafts
- An array of currently selected drafts (from Select mode) is now defined in `App.tsx`
- Add progress bar for all drafts (edits to `ItemList.tsx` and `App.css`)

## CHECKS :warning:

Please make sure you are aware of the following.

- [X] **The changes in this PR doesn't have private information

### CODE REVIEW

File: db/items.db [hunk 1]
```diff
@@ -10,14 +10,14 @@ CREATE TABLE IF NOT EXISTS Categories (
     );
 CREATE TABLE IF NOT EXISTS Drafts(
     draft_id INTEGER PRIMARY KEY,
-    sort_index INT NOT NULL AS (draft_id) STORED,
-    item_name TEXT NOT NULL,
-    item_image_filename TEXT NOT NULL,
-    category_id INT NOT NULL,
-    item_state_id INT NOT NULL,
```
Comment (kappanjoe) [inline db/items.db:17]: Removed NOT NULL to allow for incomplete drafts

File: python/main.py [hunk 2]
```diff
@@ -230,14 +279,22 @@ def get_items_list():
...
+category: str = Form(...), item_state_id: str = Form(...), description: str = Form(...),
+delivery_id: str = Form(...), price: str = Form(...)):
+    
+    # Raise exception if image file extension is not .jpg
+    if not image.filename.endswith(".jpg"):
+        raise HTTPException(status_code=400, detail="Image filename does not end with .jpg")
+    
+    image_title = image.filename.split(".")[0].encode()
+    hashed_filename = hashlib.sha256(image_title).hexdigest() + ".jpg"
+
```
Comment (kappanjoe) [inline python/main.py:294]: Added filename hashing to avoid filename character errors

File: python/main.py [hunk 3]
```diff
@@ -197,28 +222,52 @@ async def get_image(image_filename):
...
+    image = draft_images / image_filename
+
+    if not image_filename.endswith(".jpg"):
+        raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
+
+    if not image.exists():
+        logger.info(f"Image not found: {image}")
+        image = draft_images / "default.jpg"
+
+    return FileResponse(image)
```
Comment (kappanjoe) [inline python/main.py:239]: Added GET /draft_image to pull images from new draft_images folder

File: typescript/simple-mercari-web/src/App.css [hunk 4]
```diff
@@ -54,26 +54,64 @@
...
+  background-color: limegreen;
+}
+
+.IndicatorOff {
+  display: table-cell;
+  width: auto;
+  height: 6px;
+  background-color: lightgray;
+}
+
```
Comment (kappanjoe) [inline typescript/simple-mercari-web/src/App.css:114]: Added Progress Bar!

File: typescript/simple-mercari-web/src/App.tsx [hunk 5]
```diff
@@ -12,18 +12,60 @@ function App() {
...
+    }
+  };
+  
+  function newChange(id: number, checked: boolean) {
+    if (checked) {
+      setIsCheck([...isCheck, id]);
+    } else {
+      setIsCheck(isCheck.filter(item => item !== id));
+    }
+  }
```
Comment (kappanjoe) [inline typescript/simple-mercari-web/src/App.tsx:51]: Delete Items handling

File: typescript/simple-mercari-web/src/components/ItemList/ItemList.tsx [hunk 6]
```diff
@@ -53,22 +58,48 @@ export const ItemList: React.FC<Prop> = (props) => {
...
+                <div className='ProgressTitle'>Price</div>
+              </div>
+              <div className='IndicatorRow'>
+                <div className={completeCheck(item.image)}></div>
+                <div className={completeCheck(item.category)}></div>
+                <div className={completeCheck(item.description)}></div>
+                <div className={completeCheck(item.delivery)}></div>
+                <div className={completeCheck(item.price)}></div>
+              </div>
+            </div>
```
Comment (kappanjoe) [inline typescript/simple-mercari-web/src/components/ItemList/ItemList.tsx:102]: Added Checkbox handling and Progress Bar

**Review by narufish — APPROVED** [review]
LGTM!

**Pull request merged**